packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -491,3 +493,35 @@ void _copyServiceDefinitionsManifest(List<Map<String, String>> services, File ma
   final Map<String, dynamic> json = <String, dynamic>{ 'services' : jsonServices };
   manifest.writeAsStringSync(JSON.encode(json), mode: FileMode.WRITE, flush: true);
 }
+
+Future<Null> upgradePbxProjWithFlutterAssets(String app) async {
+  final File xcodeProjectFile = fs.file(fs.path.join('ios', 'Runner.xcodeproj',
+                                                     'project.pbxproj'));
+  assert(await xcodeProjectFile.exists());
+  final List<String> lines = await xcodeProjectFile.readAsLines();
+
+  final String l1 = '		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };';
+  final String l2 = '		2D5378261FAA1A9400D5DBA9 /* flutter_assets in Resources */ = {isa = PBXBuildFile; fileRef = 2D5378251FAA1A9400D5DBA9 /* flutter_assets */; };';
+  final String l3 = '		3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; name = AppFrameworkInfo.plist; path = Flutter/AppFrameworkInfo.plist; sourceTree = "<group>"; };';
+  final String l4 = '		2D5378251FAA1A9400D5DBA9 /* flutter_assets */ = {isa = PBXFileReference; lastKnownFileType = folder; name = flutter_assets; path = Flutter/flutter_assets; sourceTree = SOURCE_ROOT; };';
+  final String l5 = '				3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */,';
+  final String l6 = '				2D5378251FAA1A9400D5DBA9 /* flutter_assets */,';
+  final String l7 = '				3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */,';
+  final String l8 = '				2D5378261FAA1A9400D5DBA9 /* flutter_assets in Resources */,';
+
+  if (lines.contains(l2))
+    return;
+
+  printStatus("Upgrading project.pbxproj of $app' to include the "
+              "'flutter_assets' directory");
+  lines.insert(lines.indexOf(l1) + 1, l2);
```

Comment:
This'll work in by far the most cases, but *if* the user has been tinkering with the project, there's a small risk they might have changed some of these, in which case it'll be hard for us to upgrade the project automatically.

Could you check if these lines exist, and print out a message telling the user what to do manually if they don't?

(if a line doesn't exist, `indexOf()` returns `-1`, so the current code will insert the corresponding new line at the beginning of the file.)